Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CalculateMonteCarloAbsorption.py

```diff
@@ -0,0 +1,775 @@
+
+                acc_ws = self._multiply(acc_1, acc_2)
+
+            elif self._shape == 'Cylinder':
+                ss_monte_carlo_alg.setProperty("InputWorkspace", container_wave_1)
+                ss_monte_carlo_alg.setProperty("InnerRadius", self._container_inner_radius)
+                ss_monte_carlo_alg.setProperty("OuterRadius", self._container_outer_radius)
+                ss_monte_carlo_alg.setProperty("Shape", "Annulus")
+                ss_monte_carlo_alg.execute()
+                acc_ws = ss_monte_carlo_alg.getProperty("OutputWorkspace").value
+
+            elif self._shape == 'Annulus':
+                ss_monte_carlo_alg.setProperty("InputWorkspace", container_wave_1)
+                ss_monte_carlo_alg.setProperty("InnerRadius", self._container_inner_radius)
+                ss_monte_carlo_alg.setProperty("OuterRadius", self._container_outer_radius)
+                ss_monte_carlo_alg.execute()
+                acc_1 = ss_monte_carlo_alg.getProperty("OutputWorkspace").value
+
+                ss_monte_carlo_alg.setProperty("InputWorkspace", container_wave_2)
+                ss_monte_carlo_alg.execute()
+                acc_2 = ss_monte_carlo_alg.getProperty("OutputWorkspace").value
+
+                acc_ws = self._multiply(acc_1, acc_2)
+
+            for log_name, log_value in container_kwargs.items():
+                sample_log_names.append("container_" + log_name.lower())
+                sample_log_values.append(log_value)
+
+            acc_ws = self._convert_from_wavelength(acc_ws)
+
+            self._add_sample_log_multiple(acc_ws, sample_log_names, sample_log_values)
+
+            if not self.isChild():
+                mtd.addOrReplace(self._acc_ws_name, acc_ws)
+
+            self._output_ws = self._group_ws([ass_ws, acc_ws])
+        else:
+            self._output_ws = self._group_ws([ass_ws])
+
+        self.setProperty('CorrectionsWorkspace', self._output_ws)
+
+    def _setup(self):
+
+        # The beam properties and monte carlo properties are simply passed straight on to the
+        # SimpleShapeMonteCarloAbsorptionCorrection algorithm so they are being put into
+        # a dictionary for simplicity
+
+        self._general_kwargs = {'BeamHeight': self.getProperty('BeamHeight').value,
+                                'BeamWidth': self.getProperty('BeamWidth').value,
+                                'NumberOfWavelengthPoints': self.getProperty('NumberOfWavelengthPoints').value,
+                                'EventsPerPoint': self.getProperty('EventsPerPoint').value,
+                                'Interpolation': self.getProperty('Interpolation').value}
+
+        self._sample_ws = self.getProperty("SampleWorkspace").value
+        self._container_ws = self.getProperty('ContainerWorkspace').value
+        self._shape = self.getProperty('Shape').value
+        self._height = self.getProperty('Height').value
+
+        self._sample_unit = self._sample_ws.getAxis(0).getUnit().unitID()
```

Comment:
`getAxis` throws an error for workspace groups, and the error message is not too nice to show to the user. Since this algorithm is not intended to run over workspace groups I would suggest have an explicit check for that in `validateInputs`, and raise a descriptive error message there. The same applies to container workspace, since L434 will complain for a workspace group.

Reply:
Would it be useful for the algorithm to be able to run sequentially across WorkspaceGroups? This could be done by skipping these setup checks when the Workspace is a WorkspaceGroup; the checks are done first for the WorkspaceGroup and then each Workspace in turn as they are sequentially used in the algorithm.

Reply:
Generally the algorithms should transparently run over groups, but in this case it is debatable, since the output itself is a group workspace. Also different workspaces are normally for different samples, so allowing to run the same absorption over a group might be error prone. So I think we should rather forbid it here, and also later in the interface tab.

Reply:
For the former point, multiple WorkspaceGroups would be created in the case of supplying a WorkspaceGroup as input.
As long as the other properties in setup are initialized per workspace in the workspace group, the latter shouldn't be an issue, as each run will be independent of the previous and subsequent runs.
The checks that appear important are whether the container and sample workspace types are matched and have the same length, when they are both supplied.